src/allotropy/parsers/moldev_softmax_pro/softmax_pro_structure.py

```diff
@@ -21,13 +22,53 @@
     try_float_or_none,
     try_int,
     try_int_or_none,
+    try_str_from_series,
+    try_str_from_series_or_none,
 )
 
 BLOCKS_LINE_REGEX = r"^##BLOCKS=\s*(\d+)$"
 END_LINE_REGEX = "~End"
 EXPORT_VERSION = "1.3"
 
 
+def is_float(value: Any) -> bool:
```

Comment:
again, this is confusingly named: 
1. I'd expect `is_float(float("NaN"))` to return True. How about `is_non_nan_float`?
2. This will return `true` for any String that can parse as a float, but I wouldn't expect `is_float('3')` to return True.

How about `can_parse_as_float_non_nan` or similar?